### PR TITLE
partial transaction history

### PR DESCRIPTION
**Pull request by yushih**
https://emurgo.atlassian.net/browse/YOEXT-669

Summary of changes:

1. Use the new "getting tx hashes for addresses" API and the existing but not-yet-used "getting full tx data by tx hashes" API to get the tx list. Originally the "getting tx data for addresses" API was used.

2. Only query and refresh the latest txs, and query for more txs on-demand when the UI needs to show more. Only 100 txs are kept in storage.

3. The interface of `TransactionsStores.js` has to be changed. Then the UI layers are changed to adapt.

4. Because the way the Ergo wallets handle txs are not changed, the original `TransactionsStore` are renamed `LegacyTransactionsStore` and becomes a member of the new `TransactionsStores`.  Ergo-related requests are delegated to it.

5. Since we don't have the full tx history, we can no longer determine if an address is used by checking if there is a tx outputting to it, an "IsUsed" boolean field is added to the "Address" table. The result of the "checking for used addresses" API is used to set its value. 

6. Totally broke StoryBook.

### CODE REVIEW

**Review by a-hosani — APPROVED** [review]

**Review by a-hosani — APPROVED** [review]

**Review by a-hosani — APPROVED** [review]

**Review by a-hosani — APPROVED** [review]

**Review by a-hosani — APPROVED** [review]

**Review by a-hosani — APPROVED** [review]

**Review by a-hosani — APPROVED** [review]

**Comment by yushih** [comment]
Should merge/release https://github.com/Emurgo/yoroi-frontend/pull/3060 first or together. Otherwise, this breaks tx exporting.

**Comment by neuodev** [comment]
[autoscript] ⚠️ `yushi/partial-tx-history` doesn't full our branch naming convention.
Should follow `<tag>/<jira-ticket-id>/<short-name>`
<details>
<summary>Details</summary>

- `<tag>`: `fix`, `feat`, `chore`, `refactor`, `test`, ...
- `<jira-ticket-id>`: The issue ID of JIRA all capital letters
- `<short-name>`: A short and descriptive name for the branch
</details>

[](__JIRA_COMMENT_FLAG)

**Comment by yushih** [comment]
Require https://github.com/Emurgo/yoroi-backend/pull/21 to be deployed.

File: packages/yoroi-extension/app/stores/toplevel/TransactionsStore.js [hunk 1]
```diff
@@ -52,32 +48,32 @@ import { RustModule } from '../../api/ada/lib/cardanoCrypto/rustLoader';
...
-     * To reduce memory footprint, this request does not store the whole transaction
-     *  list, but instead only derived data from it.
-     *
-     */
-    allRequest: CachedRequest<GetTransactionsDataFunc>,
+    // used to initially load the saved txs and then periodically refresh for
+    // new txs
+    headRequest: CachedRequest<GetTransactionsFunc>,
+    // used to "load more transactions"
+    tailRequest: CachedRequest<GetTransactionsFunc>,
```
Comment (zuzunker) [inline packages/yoroi-extension/app/stores/toplevel/TransactionsStore.js:76]: nice

**Review by zuzunker — APPROVED** [review]

**Pull request merged**